core/src/main/scala/org/apache/spark/h2o/utils/H2OContextRestAPIUtils.scala

```diff
@@ -279,7 +279,12 @@ trait H2OContextRestAPIUtils extends H2OContextUtils {
         case unknown => throw new IllegalArgumentException(s"Unsupported HTTP request type $unknown")
       }
       getCredentials(conf).foreach(creds => request.setHeader(HttpHeaders.AUTHORIZATION, creds))
-      httpClient.execute(request).getEntity.getContent
+      val result = httpClient.execute(request)
+      val statusCode = result.getStatusLine.getStatusCode
+      statusCode match {
+        case 401 => throw new Exception("Unauthorized")
+      }
```

Comment:
Wouldn't the  pattern matching throw an exception if  `statusCode` is different from `401`.

Reply:
Very good catch -> it will throw MatchError, thanks!